### ISSUE TITLE
Add a verification about the Composer version

### ISSUE THREAD

**Issue by j0k3r** [pyrech/composer-changelogs#8]
After updating `composer-changelogs` globally, I got this error when updating a repo:

```
j0k@MBP:~/Sites/f43.me$ composer up
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Updating j0k3r/graby-site-config (1.0.3 => 1.0.4)
    Checking out cf088ca2100eeec3f230cc187a5b489e61fe97f1

Writing lock file
Generating autoload files
PHP Fatal error:  Call to undefined method Composer\Package\CompletePackage::getFullPrettyVersion() in /Users/j0k/.composer/vendor/pyrech/composer-changelogs/src/OperationHandler/UpdateHandler.php on line 58
PHP Stack trace:
PHP   1. {main}() /usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar:0
PHP   2. require() /usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar:24
PHP   3. Composer\Console\Application->run($input = *uninitialized*, $output = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/bin/composer:43
PHP   4. Symfony\Component\Console\Application->run($input = *uninitialized*, $output = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/Console/Application.php:84
PHP   5. Composer\Console\Application->doRun($input = *uninitialized*, $output = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:126
PHP   6. Symfony\Component\Console\Application->doRun($input = *uninitialized*, $output = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/Console/Application.php:147
PHP   7. Symfony\Component\Console\Application->doRunCommand($command = *uninitialized*, $input = *uninitialized*, $output = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:195
PHP   8. Symfony\Component\Console\Command\Command->run($input = *uninitialized*, $output = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:874
PHP   9. Composer\Command\UpdateCommand->execute($input = *uninitialized*, $output = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:257
PHP  10. Composer\Installer->run() phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/Command/UpdateCommand.php:140
PHP  11. Composer\EventDispatcher\EventDispatcher->dispatchScript($eventName = *uninitialized*, $devMode = *uninitialized*, $additionalArgs = *uninitialized*, $flags = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/Installer.php:336
PHP  12. Composer\EventDispatcher\EventDispatcher->doDispatch($event = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:91
PHP  13. call_user_func:{phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:151}(*uninitialized*, *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:151
PHP  14. Pyrech\ComposerChangelogs\ChangelogsPlugin->postUpdate($event = *uninitialized*) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:151
PHP  15. Pyrech\ComposerChangelogs\Outputter->getOutput() /Users/j0k/.composer/vendor/pyrech/composer-changelogs/src/ChangelogsPlugin.php:76
PHP  16. Pyrech\ComposerChangelogs\Outputter->createOperationOutput($output = *uninitialized*, $operation = *uninitialized*) /Users/j0k/.composer/vendor/pyrech/composer-changelogs/src/Outputter.php:61
PHP  17. Pyrech\ComposerChangelogs\OperationHandler\UpdateHandler->getOutput($operation = *uninitialized*, $urlGenerator = *uninitialized*) /Users/j0k/.composer/vendor/pyrech/composer-changelogs/src/Outputter.php:92

Fatal error: Call to undefined method Composer\Package\CompletePackage::getFullPrettyVersion() in /Users/j0k/.composer/vendor/pyrech/composer-changelogs/src/OperationHandler/UpdateHandler.php on line 58

Call Stack:
    0.0116     410496   1. {main}() /usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar:0
    0.0120     412248   2. require('phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/bin/composer') /usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar:24
    0.0587    3501888   3. Composer\Console\Application->run(???, ???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/bin/composer:43
    0.0612    3818560   4. Symfony\Component\Console\Application->run(???, ???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/Console/Application.php:84
    0.0622    3943288   5. Composer\Console\Application->doRun(???, ???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:126
    0.0635    4065184   6. Symfony\Component\Console\Application->doRun(???, ???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/Console/Application.php:147
    0.0638    4066104   7. Symfony\Component\Console\Application->doRunCommand(???, ???, ???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:195
    0.0638    4066576   8. Symfony\Component\Console\Command\Command->run(???, ???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:874
    0.0647    4070960   9. Composer\Command\UpdateCommand->execute(???, ???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:257
    0.3706    9571456  10. Composer\Installer->run() phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/Command/UpdateCommand.php:140
  135.7863  256004064  11. Composer\EventDispatcher\EventDispatcher->dispatchScript(???, ???, ???, ???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/Installer.php:336
  135.7863  256004504  12. Composer\EventDispatcher\EventDispatcher->doDispatch(???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:91
  135.7957  256110168  13. call_user_func:{phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:151}(???, ???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:151
  135.7957  256110280  14. Pyrech\ComposerChangelogs\ChangelogsPlugin->postUpdate(???) phar:///usr/local/Cellar/composer/1.0.0-alpha10/libexec/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php:151
  135.7957  256110352  15. Pyrech\ComposerChangelogs\Outputter->getOutput() /Users/j0k/.composer/vendor/pyrech/composer-changelogs/src/ChangelogsPlugin.php:76
  135.7957  256110880  16. Pyrech\ComposerChangelogs\Outputter->createOperationOutput(???, ???) /Users/j0k/.composer/vendor/pyrech/composer-changelogs/src/Outputter.php:61
  135.7958  256112040  17. Pyrech\ComposerChangelogs\OperationHandler\UpdateHandler->getOutput(???, ???) /Users/j0k/.composer/vendor/pyrech/composer-changelogs/src/Outputter.php:92

j0k@MBP:~/Sites/f43.me$
```

I checked my composer version (which was installed using homebrew) and I got this version: `Composer version 1.0.0-alpha10 2015-04-14 21:18:51` which seems quite old **but** it's the latest release.

After upgrading composer, I got a newer version `Composer version 1.0-dev (7a9eb02190d334513e99a479510f87eed18cf958) 2015-10-07 10:05:33` and no more error when performing a `composer up`.

Maybe you should add a verification on the Composer version to see what you can or cannot do.

**Comment by pyrech** [comment]
Oups, you're right, I didn't see that this method was added in https://github.com/composer/composer/commit/99dab8aebdef0f833294268af026069f6f1368cf (merged July 4th).

A fix shouldn't be too hard to do :)

**Comment by pyrech** [comment]
The fix was just released in 1.1.1

I tried to test for quite old versions of composer but the only version I could simply get working in tests was alpha-10 (I need to instantiate some objects whose public interface changes for tests).

Anyway after a lot of manual tests, I think it should work for alpha7 to alpha-10 ;) Thank you!

**Comment by j0k3r** [comment]
Oh waouw, thank you !
